### PR TITLE
Fix token refresh

### PR DESCRIPTION
**Pull request by maxlaverse**
* Stores the Token between calls
* Add tests

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/kubernetes/credentials/OpenShiftBearerTokenCredentialImpl.java [hunk 1]
```diff
@@ -5,170 +5,151 @@
...
+        String error = "", errorDescription = "";
+        Token token = new Token();
+        for (NameValuePair pair : pairs) {
+            switch (pair.getName()) {
+                case "access_token":
+                    token.value = pair.getValue();
+                    break;
+                case "expires_in":
+                    try {
+                        token.expire = System.currentTimeMillis() + Long.parseLong(pair.getValue()) * 1000 - EARLY_EXPIRE_DELAY;
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/kubernetes/credentials/OpenShiftBearerTokenCredentialImpl.java]: is 100ms enough here - what if I wanted to grab this for use in a build in a pipeline - there could be a few minutes delay before starting a script that needs the credential and the use of the credential?

Comment (maxlaverse) [reply src/main/java/org/jenkinsci/plugins/kubernetes/credentials/OpenShiftBearerTokenCredentialImpl.java]: You're right. I overlooked it thought it was seconds. I don't know what a good value would be but 5 minutes sound reasonnable.

**Comment by maxlaverse** [comment]
@jtnord Do you have an Openshift installation do test it ? 
I want to add some integration tests to fake the oauth server but I don't think I'll have time today.

**Comment by maxlaverse** [comment]
I will have to check if `getToken()` is sometimes called with different URLs :/

**Comment by maxlaverse** [comment]
Thanks for the review @jtnord

**Pull request merged**